### PR TITLE
Wrap endblock

### PR DESCRIPTION
**Pull request by RedBird96**
## Purpose of Changes and their Description

Implemented a wrapper function for EndBlocker to catch and handle any potential panics

## Are these changes tested and documented?

- [ ] If tested, please describe how. If not, why tests are not needed.
- [ ] If documented, please describe where. If not, describe why docs are not needed.
- [ ] Added to `Unreleased` section of `CHANGELOG.md`?

## Still Left Todo

*Fill this out if this is a Draft PR so others can help.*

### CODE REVIEW

File: x/emissions/module/module.go [hunk 1]
```diff
@@ -117,6 +117,14 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.Raw
 
 // EndBlock returns the end blocker for the emissions module.
 func (am AppModule) EndBlock(ctx context.Context) error {
+	defer func() {
+		sdkCtx := sdk.UnwrapSDKContext(ctx)
+		if r := recover(); r != nil {
+			err := fmt.Errorf("recovered from panic in EndBlocker: %v", r)
+			sdkCtx.Logger().Error("Error Getting module params", err)
```
Comment (kpeluso) [inline x/emissions/module/module.go:124]: Wrong error message!

**Review by kpeluso — CHANGES_REQUESTED** [review]
Simple + elegant! Just 1 small detail to fix.

File: x/emissions/module/module.go [hunk 2]
```diff
@@ -117,6 +117,14 @@ func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.Raw
...
+		}
+	}()
+
 	err := EndBlocker(ctx, am)
 	if err != nil {
 		sdkCtx := sdk.UnwrapSDKContext(ctx)
```
Comment (relyt29) [inline x/emissions/module/module.go:130]: can we fix this typo in this PR too: `EnbBlocker`

Comment (relyt29) [reply x/emissions/module/module.go:130]: (it's the line below this one, won't let me leave a comment on just that line)

**Comment by relyt29** [comment]
I am confused how this PR works, is this just going to catch panics on unwrapping the sdk context?

**Review by relyt29 — APPROVED** [review]
I don't think the sdk context will be a real problem

**Review by kpeluso — APPROVED** [review]

**Pull request merged**